packages/dev/buildTools/src/generateDeclaration.ts

```diff
@@ -37,7 +37,14 @@ function getModuleDeclaration(
 ) {
     const distPosition = filename.indexOf("/dist");
     const packageVariables = getPackageMappingByDevName(config.devPackageName);
-    const moduleName = getPublicPackageName(packageVariables[buildType], filename) + filename.substring(distPosition + 5).replace(".d.ts", "");
+    let extraPackageName = filename
+        .substring(distPosition + 5)
+        .replace(".d.ts", "")
+        .replace(/\\/g, "/");
+    if (extraPackageName.includes("/dist/")) {

```

Comment:
I can't quite understand what is happening here. It looks like the previous line is getting the substring after `/dist`, but then this line is trying to account for the possibility that there is still a `/dist/` in the string? How can that happen, or am I misreading this?

Reply:
I pushed an experiment from the windows machine and didn't push the changes from the mac. Updated now.